dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -0,0 +1,756 @@
+        if (snappyTest == null) {
+            snappyTest = new SnappyTest();
+            snappyTest.generateConfig("locators");
+            snappyTest.generateConfig("servers");
+            snappyTest.generateConfig("leads");
+        }
+    }
+
+    public static synchronized void HydraTask_generateSnappyConfig() {
+        locatorsFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "locators";
+        serversFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "servers";
+        leadsFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "leads";
+        try {
+            String dirPath = snappyTest.getLogDir();
+            File locatorsFile = new File(locatorsFilePath);
+            File serversFile = new File(serversFilePath);
+            File leadsFile = new File(leadsFilePath);
+            if (dirPath.contains("locator")) {
+                String locatorLogDir = localHost + " -dir=" + dirPath;
+                if (!locatorsFileContent.contains(locatorLogDir)) {
+                    locatorsFileContent.add(locatorLogDir);
+                    snappyTest.writeToFile(locatorLogDir, locatorsFile);
+                }
+            } else if (dirPath.contains("Store") || dirPath.contains("server")) {
+                String serverLogDir = localHost + " -dir=" + dirPath;
+                if (!serversFileContent.contains(serverLogDir)) {
+                    serversFileContent.add(serverLogDir);
+                    snappyTest.writeToFile(serverLogDir, serversFile);
+                }
+            } else if (dirPath.contains("lead")) {
+                String leadLogDir = localHost + " -dir=" + dirPath;
+                if (leadHost == null) {
+                    leadHost = localHost;
+                }
+                Log.getLogWriter().info("Lead host is: " + leadHost);
+                if (!leadsFileContent.contains(leadLogDir)) {
+                    leadsFileContent.add(leadLogDir);
+                    snappyTest.writeToFile(leadLogDir, leadsFile);
+                }
+            }
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getStackTrace());
+        }
+    }
+
+    protected void writeToFile(String logDir, File file) {
+        try {
+            FileWriter fw = new FileWriter(file.getAbsoluteFile(), true);
+            BufferedWriter bw = new BufferedWriter(fw);
+            bw.write(logDir);
+            bw.newLine();
+            bw.close();
+        } catch (IOException e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+    }
+
+    /**
+     * Executes user scripts for InitTask.
+     */
+    public static void HydraTask_executeSQLScriptsInInitTask() {
+        Vector scriptNames, paramList = null;
+        scriptNames = SnappyPrms.getSQLScriptNamesForInitTask();
+        paramList = SnappyPrms.getSQLScriptParamsForInitTask();
```

Comment:
above statement will return null values if corresponding conf variables are not declared by user. please throw an  exception if these varaibles are not declared while reading inside the SnappyPrms